src/EnergyPlus/SolarShading.cc

```diff
@@ -586,11 +586,11 @@ void GetShadowingInput(EnergyPlusData &state)
         state.dataIPShortCut->cAlphaArgs(aNum) = "No";
         state.dataSysVars->ReportExtShadingSunlitFrac = false;
     }
-    int ExtShadingSchedNum;
     if (state.dataSysVars->shadingMethod == ShadingMethod::Imported) {
+        int ExtShadingSchedNum;
         for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
             ExtShadingSchedNum = ScheduleManager::GetScheduleIndex(state, state.dataSurface->Surface(SurfNum).Name + "_shading");
-            if (ExtShadingSchedNum) {
+            if (ExtShadingSchedNum != 0) {
```

Comment:
I'm not actually sure this has any change.  C++ is pretty [clear](https://en.cppreference.com/w/cpp/language/implicit_conversion#Boolean_conversions) about what happens to integers when they are cast to a boolean for this type of purpose, and 0 is going to be false while any other integer will be true.  That's the same thing you are doing by adding the `!= 0` operation here.

I made a small example to confirm that it behaves the same either way [here](https://godbolt.org/z/hsf6zcjGM).  

All that said, it is perfectly acceptable to explicitly check against the integer zero value, to avoid the implicit cast to boolean.